packages/react-jss/src/withStyles.js

```diff
@@ -79,6 +77,61 @@ export default function withStyles<Theme: {}, S: Styles<Theme>>(
       // $FlowFixMe
       static defaultProps = {...InnerComponent.defaultProps}
 
+      static getSheetClasses(sheet, dynamicRules: ?DynamicRules) {
+        const classes = {}
+
+        // $FlowFixMe
+        for (const key in sheet.styles) {
+          classes[key] = sheet.classes[key]
+
+          if (dynamicRules && key in dynamicRules) {
+            classes[key] += ` ${sheet.classes[dynamicRules[key].key]}`
+          }
+        }
+
+        return classes
+      }
+
+      static updateDynamicRules(props: HOCProps<Theme, Props>, {dynamicRules, sheet}: State) {
+        if (!sheet) {
+          return
+        }
+
+        for (const key in dynamicRules) {
+          // $FlowFixMe: Not sure why it throws an error here
+          sheet.update(dynamicRules[key].key, props, {})
```

Comment:
why is the last argument empty object? You want to avoid plugins execution on the return values?

Reply:
I think flow also complained here because it couldn't figure which overload to use I believe.